lib/jsdom/events/EventTarget.js

```diff
@@ -0,0 +1,188 @@
+﻿"use strict";
+
+var MultihandleWeakMap = require("../MultihandleWeakMap");
+var EventException = require("./EventException").EventException;
+
+var instances = new MultihandleWeakMap();
+
+function EventTarget() {
+  instances.set(this, {
+    events: {}
+  });
+}
+
+EventTarget.prototype.addEventListener = function (type, callback, capture) {
+  type = String(type);
+  if (callback === null || callback === undefined || typeof callback !== "function") {
+    return;
+  }
+  if (capture === undefined || capture === null) {
+    capture = false;
+  }
+  capture = Boolean(capture);
+
+  var props = instances.get(this);
+
+  if (props.events[type] === undefined) {
+    props.events[type] = {
+      capturing: [],
+      bubbling: []
+    };
+  }
+
+  var handlers = props.events[type][capture ? "capturing" : "bubbling"];
+  if (handlers.indexOf(callback) !== -1) {
+    return;
+  }
+
+  handlers.push(callback);
+};
+
+EventTarget.prototype.removeEventListener = function (type, callback, capture) {
+  type = String(type);
+  if (callback === null || callback === undefined || typeof callback !== "function") {
+    return;
+  }
+  if (capture === undefined || capture === null) {
+    capture = false;
+  }
+  capture = Boolean(capture);
+
+  var props = instances.get(this);
+  if (!props.events[type]) {
+    return;
+  }
+
+  var handlers = props.events[type][capture ? "capturing" : "bubbling"];
+  var idx = handlers.indexOf(callback);
+
+  if (idx !== -1) {
+    handlers.splice(idx, 1);
+  }
+};
+
+EventTarget.prototype.dispatchEvent = function (event) {
+  if (event === null || event === undefined) {
+    throw new EventException(0, "Null event");
+  }
+  if (event._type === null || event._type === "") {
+    throw new EventException(0, "Uninitialized event");
+  }
+
+  var targetList = [];
+
+  event._target = this;
+
+  //per the spec we gather the list of targets first to ensure
+  //against dom modifications during actual event dispatch
+  var target = this,
+      targetParent = target._parentNode;
+  while (targetParent) {
+    targetList.push(targetParent);
+    target = targetParent;
+    targetParent = target._parentNode;
+  }
+  targetParent = target.defaultView;
+  if (targetParent) {
+    targetList.push(targetParent);
+  }
+
+  var iterator = backwardIterator(targetList);
+
+  event._eventPhase = event.CAPTURING_PHASE;
+  dispatchPhase(event, iterator);
+
+  iterator = singleIterator(event._target);
+  event._eventPhase = event.AT_TARGET;
+  dispatchPhase(event, iterator);
+
+  if (event._bubbles) {
+    iterator = forwardIterator(targetList);
+    event._eventPhase = event.BUBBLING_PHASE;
+    dispatchPhase(event, iterator);
+  }
+
+  event._currentTarget = null;
+  event._eventPhase = event.NONE;
+
+  return !event._preventDefault;
+};
+
+module.exports.EventTarget = EventTarget;
```

Comment:
Hopefully with the WeakMap -> Symbol move, addAlias can go away, and this can become just `module.exports = EventTarget`